includes/utils/class-amp-carousel.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Class AMP_Carousel.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Carousel
+ *
+ * Gets the markup for an <amp-carousel>.
+ *
+ * @since 1.4.1
+ */
+class AMP_Carousel {
+
+	/**
+	 * Value used for width of amp-carousel.
+	 *
+	 * @var int
+	 */
+	const FALLBACK_WIDTH = 600;
+
+	/**
+	 * Value used for height of amp-carousel.
+	 *
+	 * @var int
+	 */
+	const FALLBACK_HEIGHT = 480;
+
+	/**
+	 * An object representation of the DOM.
+	 *
+	 * @var DOMDocument
+	 */
+	protected $dom;
+
+	/**
+	 * Instantiates the class.
+	 *
+	 * @param DOMDocument $dom The dom to use to create a carousel.
+	 */
+	public function __construct( $dom ) {
+		$this->dom = $dom;
+	}
+
+	/**
+	 * Creates and gets an <amp-carousel> with the given images and captions.
+	 *
+	 * @param array[] $images_and_captions {
+	 *     The images and their respective captions, if any.
+	 *
+	 *     @type DOMElement  $image A representation of an image, in index 0.
+	 *     @type string|null $caption The caption for that image, if any, in index 1.
+	 * }
+	 * @return DOMElement A representation of the <amp-carousel>.
+	 */
+	public function create_and_get( $images_and_captions ) {
+		$images = [];
```

Comment:
This function is mainly moved from `AMP_Gallery_Block_Sanitizer`, with few changes.